File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    stack = []
    answer = 0
    for move in moves:
        for i in range(len(board)):
            v = board[i][move-1]
            if v != 0:
                if stack and stack[-1] == v:
                    stack.pop()
                    answer += 2
                else:
                    stack.append(v)
                board[i][move-1] = 0
                break
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.Arrays;
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int[] arrIdx = new int[board[0].length];
        Arrays.fill(arrIdx, board.length);

        for (int i = 0; i < board[0].length; i++) {
            for (int j = 0; j < board.length; j++) {
                if (board[j][i] != 0 && arrIdx[i] == board.length) {
                    arrIdx[i] = j;
                    break;
                }
            }
        }

        System.out.println(Arrays.toString(arrIdx));

        Stack<Integer> stack = new Stack<>();
        int answer = 0;
        for (int move : moves) {
            int idx = arrIdx[move - 1];
            if(idx == board.length) {
                continue;
            }

            arrIdx[move - 1]++;
            int v = board[idx][move - 1];

            if (!stack.isEmpty() && stack.peek() == v) {
                stack.pop();
                answer += 2;
            } else {
                stack.push(v);
            }

        }

        return answer;
    }
}